source/hic_hal/freescale/kl26z/uart.c

```diff
@@ -283,9 +237,18 @@ void UART_RX_TX_IRQHandler(void)
         if ((s1 & UART_S1_NF_MASK) || (s1 & UART_S1_FE_MASK)) {
             errorData = UART->D;
         } else {
-            read_buffer.data[read_buffer.idx_in++] = UART->D;
-            read_buffer.idx_in &= (UART_BUFFER_SIZE - 1);
-            read_buffer.cnt_in++;
+            uint32_t free;
+            uint8_t data;
+            
+            data = UART1->D;
+            free = circ_buf_count_free(&read_buffer);
+            if (free > RX_OVRF_MSG_SIZE) {
+                circ_buf_push(&read_buffer, data);
+            } else if (RX_OVRF_MSG_SIZE == free) {
+                circ_buf_write(&read_buffer, (uint8_t*)RX_OVRF_MSG, RX_OVRF_MSG_SIZE);
```

Comment:
is this msg going to host or target? sounds little dangerous to send any data even overflow happens, e.g if there is some protocol used in serial port this might causes confusion&trouble. Instead maybe we could create fail.txt file to fs with similar message which indicate that there was overflow during data transfer? protocol anyway need somekind of validation so even some data was dropped probably it is already notified by tools/application.. but at least I propose to add line feeds to message, covers situation where protocol is simple line based. then it is handled proper way at least by our test tools..